src/frontend/apps/lti_site/components/VideoWidgetProvider/widgets/SharedLiveMedia/index.spec.tsx

```diff
@@ -372,4 +372,125 @@ describe('<SharedLiveMedia />', () => {
+    const mockedSharedLiveMedia = sharedLiveMediaMockFactory({
+      title: 'Title of the file',
+      video: videoId,
+    });
+    const mockedVideo = videoMockFactory({
+      id: videoId,
+      shared_live_medias: [mockedSharedLiveMedia],
+    });
+    mockUseUploadManager.mockReturnValue({
+      addUpload: jest.fn(),
+      resetUpload: jest.fn(),
+      uploadManagerState: {},
+    });
+    useSharedLiveMedia.getState().addResource(mockedSharedLiveMedia);
+
+    render(
+      wrapInVideo(
+        <InfoWidgetModalProvider value={null}>
+          <DeleteSharedLiveMediaModalProvider value={null}>
+            <SharedLiveMediaModalWrapper />
+            <SharedLiveMedia isLive={false} isTeacher />
+          </DeleteSharedLiveMediaModalProvider>
+        </InfoWidgetModalProvider>,
+        mockedVideo,
+      ),
+    );
+
+    screen.getByRole('button', { name: 'Upload a presentation support' });
+    screen.getByRole('button', {
+      name: 'Click on this button to stop allowing students to download this media.',
+    });
+    screen.getByRole('button', {
+      name: 'Click on this button to delete the media.',
+    });
+
+    expect(
+      screen.queryByRole('button', { name: 'Share' }),
+    ).not.toBeInTheDocument();
+  });
+
+  it('renders the component in student VOD mode', () => {
+    const videoId = faker.datatype.uuid();
+    const mockedSharedLiveMedia = sharedLiveMediaMockFactory({
+      title: 'Title of the file',
+      video: videoId,
+    });
+    const mockedVideo = videoMockFactory({
+      id: videoId,
+      shared_live_medias: [mockedSharedLiveMedia],
+    });
+    mockUseUploadManager.mockReturnValue({
+      addUpload: jest.fn(),
+      resetUpload: jest.fn(),
+      uploadManagerState: {},
+    });
+    useSharedLiveMedia.getState().addResource(mockedSharedLiveMedia);
+
+    render(
+      wrapInVideo(
+        <InfoWidgetModalProvider value={null}>
+          <DeleteSharedLiveMediaModalProvider value={null}>
+            <SharedLiveMediaModalWrapper />
+            <SharedLiveMedia isLive={false} isTeacher={false} />
+          </DeleteSharedLiveMediaModalProvider>
+        </InfoWidgetModalProvider>,
+        mockedVideo,
+      ),
+    );
+
+    expect(
+      screen.queryByRole('button', { name: 'Upload a presentation support' }),
+    ).not.toBeInTheDocument();
+    expect(
+      screen.queryByRole('button', {
+        name: 'Click on this button to stop allowing students to download this media.',
+      }),
+    ).not.toBeInTheDocument();
+    expect(
+      screen.queryByRole('button', { name: 'Share' }),
+    ).not.toBeInTheDocument();
+    screen.getByRole('button', { name: 'Download all available supports' });
+    screen.getByRole('link', { name: 'Title of the file' });
+  });
+
+  it('disables the "download all" button if the media is not available to download', () => {
+    const videoId = faker.datatype.uuid();
+    const mockedSharedLiveMedia = sharedLiveMediaMockFactory({
```

Comment:
You should add multiple shared live media to test it correctly with at least one available to be downloaded.